sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-supplier.service.ts

```diff
@@ -0,0 +1,186 @@
+   * @param casSuppliers pending CAS suppliers.
+   * @param summary log summary.
+   * @param auth CAS auth details.
+   * @returns a number of update records.
+   */
+  private async requestCASAndUpdateSuppliers(
+    casSuppliers: CASSupplier[],
+    summary: ProcessSummary,
+    auth: CASAuthDetails,
+  ) {
+    let suppliersUpdated = 0;
+    for (const casSupplier of casSuppliers) {
+      summary.info(`Requesting info for CAS supplier id ${casSupplier.id}.`);
+      let supplierResponse = null;
+      try {
+        supplierResponse = await this.casService.getSupplierInfoFromCAS(
+          auth.access_token,
+          casSupplier.student.sinValidation.sin,
+          casSupplier.student.user.lastName.toUpperCase(),
+        );
+      } catch (error: unknown) {
+        summary.error("Unexpected error while requesting supplier.", error);
+      }
+      if (supplierResponse.items.length > 0) {
+        const [supplierInfo] = supplierResponse.items;
+        summary.info("Updating CAS supplier table.");
+        try {
+          const updateResult = await this.updateCASSupplier(
+            casSupplier,
+            supplierInfo,
+            SupplierStatus.Verified,
+          );
+          if (updateResult.affected) {
+            suppliersUpdated++;
+          }
+        } catch (error: unknown) {
+          summary.error("Unexpected error.", error);
+        }
+      } else {
+        summary.info("No supplier found on CAS.");
+      }
+    }
+    return suppliersUpdated;
+  }
+
+  /**
+   * Updates CAS supplier table.
+   * @param casSupplier CAS supplier to be updated.
+   * @param casSupplierResponseItem CAS supplier response item from CAS request.
+   * @param supplierStatus CAS supplier status to be updated.
+   * @returns update result.
+   */
+  async updateCASSupplier(
+    casSupplier: CASSupplier,
+    casSupplierResponseItem: CASSupplierResponseItem,
+    supplierStatus: SupplierStatus,
+  ): Promise<UpdateResult> {
+    // When multiple exists, only the active one should be saved.
+    // We will not be saving the array received at this moment, only a single entry from the received list should be persisted as JSONB.
+    const activeSupplierAddress = casSupplierResponseItem.supplieraddress.find(
+      (address) => address.status === CAS_SUPPLIER_ADDRESS_ACTIVE_STATUS,
+    );
+    let supplierAddressToUpdate: SupplierAddress = null;
+    if (activeSupplierAddress) {
+      supplierAddressToUpdate = {
+        supplierSiteCode: activeSupplierAddress.suppliersitecode,
+        addressLine1: activeSupplierAddress.addressline1,
+        city: activeSupplierAddress.city,
+        provinceState: activeSupplierAddress.province,
+        country: activeSupplierAddress.country,
+        postalCode: activeSupplierAddress.postalcode,
+        status: activeSupplierAddress.status,
+        siteProtected: activeSupplierAddress.siteprotected,
+        lastUpdated: new Date(activeSupplierAddress.lastupdated),
```

Comment:
Is the `activeSupplierAddress.lastupdated` will be properly persisted on DB. I believe that it is returning the date in PST/PDT without timezone which will be saved as UTC on DB.

Reply:
@ninosamson confirmed that verbally in a meeting.